gmso/core/topology.py

```diff
@@ -551,6 +551,133 @@ def is_typed(self, updated=False):
             self._typed = False
         return self._typed
 
+    def is_fully_typed(self, updated=False,
+                       group='topology'):
+        """Check if the topology certain group of objects of topology is fully or partially typed
+
+        Parameters
+        ----------
+        updated : bool, optional, default=False
+            If False, will update the atom type and connection type list of the
+            Topology before the checking step.
+        group : str, optional, default='topology'
+            Specific object to be checked. Options include:
+            'topology'  : check for status of the topology object
+            'sites'     : check for status of all topology.sites
+            'bonds'     : check for status of all topology.bonds
+            'angles'    : check for status of all topology.angles
+            'dihedrals' : check for status of all topology.dihedrals
+            'impropers' : check for status of all topology.impropers
+        full : bool, optional, default=True
+            If True, check if the checked group s fully typed,
+            else, check if the group is at least partially typed.
+
+        Returns
+        -------
+        bool
+            Status of the check
+        Notes
+        -----
+        `self._type` is set to True as long as the Topology is at least
+        partially typed.
+        """
+        if not updated:
+            self.update_connection_types()
+            self.update_atom_types()
+
+        typed_status = {
+            'sites' : all(site.atom_type for site in self.sites),
+            'bonds' : all(bond.bond_type for bond in self.bonds),
+            'angles' : all(angle.angle_type for angle in self.angles),
+            'dihedrals' : all(dihedral.dihedral_type for dihedral in self.dihedrals),
+            'impropers' : all(improper.improper_type for improper in self.impropers)
+    }
+
+        if group == 'topology':
+            result = list()
+            for subgroup in typed_status:
+                result.append(typed_status[subgroup])
+            return all(result)
+        elif group in typed_status:
+            return typed_status[group]
```

Comment:
```suggestion
        elif group in typed_status:
            return typed_status[group](self)
```

Reply:
The three suggestions above provide a delayed calculation of the group typing information. This way we will not be calculating any unnecessary group unless necessary.